plugins/org.eclipse.papyrus.uml.interaction.model/src/org/eclipse/papyrus/uml/interaction/internal/model/spi/impl/DefaultLayoutConstraints.java

```diff
@@ -52,6 +64,60 @@ public int getYOffset(Compartment shapeCompartment) {
 		return standardYOffsets.getOrDefault(shapeCompartment.getType(), ZERO).intValue();
 	}
 
+	@Override
+	public int getHeight(View view) {
+		return getHeight(view.getType());
+	}
+
+	@Override
+	public int getHeight(String viewType) {
+		return getHeight(viewType, NO_MODIFIER);
+	}
+
+	@Override
+	public int getHeight(View view, String modifier) {
+		return getHeight(view.getType(), modifier);
+	}
+
+	@Override
+	public int getHeight(String viewType, String modifier) {
+		return standardHeights.getOrDefault(viewType + modifier, ZERO).intValue();
+	}
+
+	@Override
+	public int getWidth(View view) {
+		return getWidth(view.getType());
+	}
+
+	@Override
+	public int getWidth(String viewType) {
+		return getWidth(viewType, NO_MODIFIER);
+	}
+
+	@Override
+	public int getWidth(View view, String modifier) {
+		return getWidth(view.getType(), modifier);
+	}
+
+	@Override
+	public int getWidth(String viewType, String modifier) {
+		return standardWidths.getOrDefault(viewType + modifier, ZERO).intValue();
```

Comment:
Would it be better if the `modifier` didn't have to include the leading underscore, but that we have a helper to concatenate non-empty modifiers with the underscore?